Code/Engine/RendererCore/Lights/Implementation/ClusteredDataExtractor.cpp

```diff
@@ -409,11 +415,16 @@ void ezClusteredDataExtractor::FillItemListAndClusterData(ezClusteredDataCPU* pD
   const ezUInt32 uiNumReflectionProbes = m_TempReflectionProbeData.GetCount();
   const ezUInt32 uiMaxReflectionProbeBlockIndex = (uiNumReflectionProbes + 31) / 32;
 
+  const ezUInt32 uiWorstCase = ezMath::Max(uiNumLights, uiNumDecals, uiNumReflectionProbes);
   for (ezUInt32 i = 0; i < NUM_CLUSTERS; ++i)
   {
-    ezUInt32 uiOffset = m_TempClusterItemList.GetCount();
+    const ezUInt32 uiOffset = m_TempClusterItemList.GetCount();
     ezUInt32 uiLightCount = 0;
 
+    // We expand m_TempClusterItemList by the worst case this loop can produce and then cut it down again to the actual size once we have filled the data. This makes sure we do not waste time on boundary checks or out of line calls compared to PushBack or PushBackUnchecked.
```

Comment:
Nice! The comment is a bit confusing, especially the last part. Maybe: "... or out of line calls **like** PushBack ..."

Reply:
Thanks, fixed!